src/service/ProductService.ts

```diff
@@ -0,0 +1,23 @@
+import { http } from '../client/httpClient';
+import { IProduct } from '../types/shoppingCart';
+
+const BASE_URL = '/products';
+
+export function ProductService() {
+  const findAllProducts = async (): Promise<IProduct[]> => {
+    const { data } = await http.get(BASE_URL);
+
+    return data;
+  };
+
+  const findProductOne = async (id: number): Promise<IProduct> => {
+    const { data } = await http.get(`${BASE_URL}/${id}`);
+
+    return data;
+  };
+
+  return {
+    findAllProducts,
+    findProductOne
+  };
+}
```

Comment:
```suggestion
export const findAllProducts = async (): Promise<IProduct[]> => {
  const { data } = await http.get(BASE_URL);

  return data;
};

export const findProductOne = async (id: number): Promise<IProduct> => {
  const { data } = await http.get(`${BASE_URL}/${id}`);

  return data;
};

```

이렇게 하는 방법도 있는데, 위와 같이 하신 이유가 뭔지 궁금합니다 😸

Reply:
여기서 사용될 함수들은 `ProductService` 라는 함수에서 import 해온 것이다 라는걸 좀 더 강조를 해보고 싶어서 했는데 해놓고 보니 굳이..? 라는 생각이 드네요🤔

리뷰어 님이 보여주신 방법이 더 깥끔하고 코드량도 줄어들 것 같아서 저렇게 변경해둘까 합니다!